File: POTD/May'24/Easy/Python/Largest_positive_integer_that_exists_with_it's_negative.py
from typing import List
class Solution:
    def findMaxK(self, nums: List[int]) -> int:
        mp = set()
        for x in nums:
            if x < 0:
                mp.add(x)
        
        ans = -1
        for x in nums:
            if x > ans:
                if -x in mp:
                    ans = max(ans,x)
        return ans

File: POTD/May'24/C++/Easy/Largest_positive_integer_that_exists_with_it's_negative.cpp
class Solution {
public:
    int findMaxK(vector<int>& nums) {
        unordered_map<int,int> mp;
        for(auto x: nums){
            if(x < 0)
                mp[x]++;
        }
        int ans = -1;
        for(auto x: nums){
            if(x > ans)
                if(mp.find(-x) != mp.end())
                    ans = max(ans,x);
        }
        return ans;
    }
};